lib/context.ts

```diff
@@ -0,0 +1,37 @@
+import { EOL } from 'node:os';
+import editorconfig from 'editorconfig';
+
+/**
+ * Context about the current invocation of the program, like what end-of-line
+ * character to use.
+ */
+export interface Context {
+  endOfLine: string;
+}
+
+export function getContext() {
+  return {
+    endOfLine: getEndOfLine(),
+  };
+}
+
+/**
+ * Gets the end of line string while respecting the `.editorconfig` and falling
+ * back to `EOL` from `node:os`.
+ */
+export function getEndOfLine() {
```

Comment:
I moved the `getEndOfLine` function here from the `string.ts` file. I think it this is a better spot for it now, since it is only used in one place now in the codebase.

Reply:
How about moving to an `editor-config.ts` file?